drivers/fpga/fpga_altera_agilex_bridge.c

```diff
@@ -0,0 +1,433 @@
+		(struct fpga_config_status *)priv_data.mbox_response_data);
+
+	k_free(priv_data.mbox_response_data);
+
+	return ret;
+}
+
+static int32_t socfpga_bridges_reset(uint32_t enable)
+{
+	uint32_t smc_cmd[2] = {0};
+	int ret = 0;
+	struct sip_svc_private_data priv_data;
+
+	/* Initialize the semaphore */
+	k_sem_init(&(priv_data.smc_sem), 0, 1);
+
+	smc_cmd[SMC_REQUEST_A2_INDEX] = FIELD_GET(BIT(0), enable);
+
+	smc_cmd[SMC_REQUEST_A2_INDEX] |= BIT(1);
+	smc_cmd[SMC_REQUEST_A3_INDEX] = BRIDGE_MASK;
+
+	ret = smc_send(SIP_SVC_PROTO_CMD_SYNC, SMC_FUNC_ID_SET_HPS_BRIDGES, smc_cmd, &priv_data);
+	if (ret) {
+		LOG_ERR("Failed to send the smc Command !!");
+		return ret;
+	}
+
+	/* Wait for the SiP SVC callback */
+	k_sem_take(&(priv_data.smc_sem), K_FOREVER);
+
+	/* Check error code */
+	if (priv_data.response.a0) {
+		ret = -ENOMSG;
+	}
+
+	return ret;
+}
+
+static int altera_fpga_on(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	int32_t ret = 0;
+
+	/* Opening SIP SVC session */
+	ret = svc_client_open();
+	if (ret) {
+		LOG_ERR("Client open Failed!");
+		return ret;
+	}
+
+	/* Check FPGA status before bridge enable/disable */
+	ret = fpga_config_ready_check();
+	if (ret) {
+		LOG_ERR("FPGA not ready. Bridge reset aborted!");
+		svc_client_close();
+		return -EIO;
+	}
+
+	/* Bridge reset */
+	ret = socfpga_bridges_reset(0x01);
+	if (ret) {
+		LOG_ERR("Bridge reset failed");
+	}
+
+	/* Ignoring the return value to return bridge reset status */
+	if (svc_client_close()) {
+		LOG_ERR("Unregistering & Closing failed");
+	}
+
+	return ret;
+}
+
+static int altera_fpga_off(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	int32_t ret = 0;
+
+	/* Opening SIP SVC session */
+	ret = svc_client_open();
+	if (ret) {
+		LOG_ERR("Client open Failed!");
+		return ret;
+	}
+
+	/* Check FPGA status before bridge enable/disable */
+	ret = fpga_config_ready_check();
+	if (ret) {
+		LOG_ERR("FPGA not ready. Bridge reset aborted!");
+		svc_client_close();
+		return -EIO;
+	}
+
+	/* Bridge reset */
+	ret = socfpga_bridges_reset(0x00);
+	if (ret) {
+		LOG_ERR("Bridge reset failed");
+	}
+
+	/* Ignoring the return value to return bridge reset status */
+	if (svc_client_close()) {
+		LOG_ERR("Unregistering & Closing failed");
+	}
+
+	return ret;
+}
+
+static int altera_fpga_init(const struct device *dev)
+{
+	if (!dev) {
+		LOG_ERR("No such device found");
+		return -ENODEV;
+	}
+
+	mailbox_smc_dev = sip_svc_get_controller("smc");
+	if (!mailbox_smc_dev) {
+		LOG_ERR("Arm SiP service not found");
+		return -ENODEV;
+	}
+
+	mailbox_client_token = sip_svc_register(mailbox_smc_dev, NULL);
+	if (mailbox_client_token == SIP_SVC_ID_INVALID) {
+		mailbox_smc_dev = NULL;
+		LOG_ERR("Mailbox client register fail");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static const struct fpga_driver_api altera_fpga_api = {
+	.on = altera_fpga_on,
+	.off = altera_fpga_off,
+};
+
+DEVICE_DT_DEFINE(DT_NODELABEL(fpga0), &altera_fpga_init, NULL, NULL, NULL, POST_KERNEL,
```

Comment:
It would be preferable to instantiate this with a macro - `DT_INST_FOREACH_STATUS_OKAY()`.

Reply:
done